sympy/solvers/ode.py

```diff
@@ -1482,6 +1491,233 @@ def _test_term(coeff, order):
     else:
         return tuple(retlist)
 
+def equivalance(max_num_pow, dem_pow):
+    # this function is made for checking the equivalance with 2F1 type of equation.
+    # We can extend it for 1F1 and 0F1 type also.
+
+    if max_num_pow == 2:
+        if dem_pow in [[2, 2], [2, 2, 2]]:
+            return "2F1"
+    elif max_num_pow == 1:
+        if dem_pow in [[1, 2, 2], [2, 2, 2], [1, 2], [2, 2]]:
+            return "2F1"
+    elif max_num_pow == 0:
+        if dem_pow in [[1, 1, 2], [2, 2], [1 ,2, 2], [1, 1], [2], [1, 2], [2, 2]]:
+            return "2F1"
+
+    return None
+
+def equivalance_hypergeometric(r, func):
+
+    # This method for finding the equivalance is only for 2F1 type.
+    # We can extend it for 1F1 and 0F1 type also.
+    f = func
+    x = func.args[0]
+    df = f.diff(x)
+    a3 = Wild('a3', exclude=[f, df, f.diff(x, 2)])
+    b3 = Wild('b3', exclude=[f, df, f.diff(x, 2)])
+    c3 = Wild('c3', exclude=[f, df, f.diff(x, 2)])
+    A = cancel(r[b3]/r[a3])
+    B = cancel(r[c3]/r[a3])
+
+    # making given equation in normal form
+    from sympy import factor, gcd
+    if B != 0:
+        I1 = factor(cancel(A.diff(x)/2 + A**2/4 - B))
+    else:
+         I1 = A
+    # computing shifted invariant(J1) of the equation
+    J1 = factor(cancel(x**2*I1 + S(1)/4))
+    num, dem = J1.as_numer_denom()
+    num = powdenest(expand(num))
+    dem = powdenest(expand(dem))
+    pow_num = set()
+    pow_dem = set()
+    # this function will compute the different powers of variable(x) in J1.
+    # then it will help in finding value of k. k is power of x such that we can express
+    # J1 = x**k * J0(x**k) then all the powers in J0 become integers.
+    def _power_counting(num):
+        _pow = {0}
+        for val in num:
+            if val.has(x):
+                if isinstance(val, Pow) and val.base == x:
+                    _pow.add(val.exp)
+                elif val == x:
+                    _pow.add(1)
+                else:
+                    _pow.update(_power_counting(val.args))
+        return _pow
+
+    pow_num = _power_counting((num, ))
+    pow_dem = _power_counting((dem, ))
+    pow_dem.update(pow_num)
+
+    _pow = pow_dem
+    k = 0
+    if _pow:
+        k = list(_pow)[0]
```

Comment:
> Is `_pow` always a 1 element set?
> 
`_pow` is a set of elements(contains all different powers of x in numerator and denominator).

> If yes then I'd rather use `(k,) = _pow` so that it raises when `_pow` is not a 1 element set.
> 
> If no then how to you know this returns the right element?

I am taking gcd of all powers of x in the numerator and denominator, which is the value of k.

Reply:
We can give multiple arguments to gcd so we can just use `gcd(*_pow)`.

Reply:
Actually, We are not taking gcd of all powers. I am taking gcd of two numbers and then the resultant gcd with another number like that. 
```
In [13]: d
Out[13]: {0, 24/7, 26/7, 4}

In [14]: d={0, S(24)/7, S(26)/7, 4}

In [15]: gcd(*S(d))
Out[15]: 1

In [16]: gcd(gcd(gcd(0, S(24)/7), S(26)/7), 4)
Out[16]: 2/7

```

Reply:
Sorry, the argument packing is incorrect. It should be:
```julia
In [60]: d={0, S(24)/7, S(26)/7, 4}                                                                                                                           

In [61]: gcd(S(d))                                                                                                                                            
Out[61]: 2/7
```